cmd/cstor-volume-mgmt/controller/volume-controller/handler.go

```diff
@@ -503,3 +515,56 @@ func IsOnlyStatusChange(oldCStorVolume, newCStorVolume *apis.CStorVolume) bool {
 	klog.Infof("No status changed for cstor volume : %s", newCStorVolume.Name)
 	return false
 }
+
+func (c *CStorVolumeController) upgrade(cv *apis.CStorVolume) (*apis.CStorVolume, error) {
+	if cv.VersionDetails.Current != cv.VersionDetails.Desired {
+		if !isCurrentVersionValid(cv) {
+			return nil, pkg_errors.Errorf("invalid current version %s", cv.VersionDetails.Current)
+		}
+		if !isDesiredVersionValid(cv) {
+			return nil, pkg_errors.Errorf("invalid desired version %s", cv.VersionDetails.Desired)
+		}
+		// Set new field DesiredReplicationFactor as ReplicationFactor
+		cv.Spec.DesiredReplicationFactor = cv.Spec.ReplicationFactor
```

Comment:
this as well should be done only if user is coming to 1.3.. not for upgrades from 1.3